packages/fhir-router/src/batch.ts

```diff
@@ -114,105 +381,78 @@ class BatchProcessor {
    * @returns The bundle entry response.
    */
   private async processBatchEntry(entry: BundleEntry): Promise<BundleEntry> {
-    this.validateEntry(entry);
-
-    const request = entry.request as BundleEntryRequest;
-
-    if (entry.resource?.resourceType && request.ifNoneExist) {
-      const searchBundle = await this.repo.search(
-        parseSearchRequest(`${entry.resource.resourceType}?${request.ifNoneExist}`)
-      );
-      const entries = searchBundle.entry as BundleEntry[];
-      if (entries.length > 1) {
-        return buildBundleResponse(badRequest('Multiple matches'));
-      }
-      if (entries.length === 1) {
-        const matchingResource = entries[0].resource as Resource;
-        if (entry.fullUrl) {
-          this.addReplacementId(entry.fullUrl, matchingResource);
-        }
-        return buildBundleResponse(allOk, matchingResource);
-      }
-
-      this.router.log('warn', 'Conditional mutation in batch', {
-        method: request.method,
-        url: request.url,
-      });
-    }
-
-    let body = entry.resource;
-    if (request.method === 'PATCH') {
-      body = this.parsePatchBody(entry);
+    const [requestPath] = splitN(entry.request?.url as string, '?', 2);
```

Comment:
We use this pattern multiple times.  1, URL manipulation always makes me nervous.  2, I wonder if we should handle this in `router.find()` or add a new `router.findByUrl()`